### ISSUE TITLE
`no-unused-modules` gives a false positive on a TypeScript declaration file

### ISSUE THREAD

**Issue by Zamiell** [benmosher/eslint-plugin-import#2127]
In TypeScript, it is common to write declaration files for 3rd-party modules or global variables.

For example:

```
// foo.d.ts
declare const foo: number;
```

This file will be flagged by the `import/no-unused-modules` rule for not having any exports found.
But this is a spurious error - this file is not intended to have any exports. The only purpose of the file is to declare the "foo" global variable.

As a quick fix, I propose that this rule ignores any file with a ".d.ts" extension, which is the signifier for a declarations file.

**Comment by proyectoramirez** [comment]
Something similar is happening to me, probably related to this. For the line
```
import process from 'node:process';
```
and the rule 
```
"import/no-unused-modules": [
  "warn",
  { "missingExports": true, "unusedExports": true }
],
```
I am getting a warning from `import/no-unused-modules`

**Comment by ljharb** [comment]
@proyectoramirez i don't think that has anything to do with typescript; does it fix the error if you import from `process` instead of `node:process`?

**Comment by proyectoramirez** [comment]
@ljharb the same error is happening to me both with and without the node protocol. It is also happening for other modules such as express (`import express from 'express';`).

**Comment by proyectoramirez** [comment]
@ljharb I am sorry for the confusion. Fixed after working around my configuration.
